RestCommClient/Classes/Media/XirsysTURNClient.m

```diff
@@ -34,109 +34,74 @@ @interface XirsysTURNClient ()
-                        NSNumber * status = [dict objectForKey:@"s"];
-                        if ([status integerValue] != 200) {
-                            NSError *responseError =
-                            [[NSError alloc] initWithDomain:[[RestCommClient sharedInstance] errorDomain]
-                                                       code:ERROR_WEBRTC_TURN
-                                                   userInfo:@{
-                                                              NSLocalizedDescriptionKey: [NSString stringWithFormat:@"Error retrieving TURN servers: %@. Check your TURN service or disable TURN altogether", [dict objectForKey:@"e"]],
-                                                              }];
-                            completionHandler(turnServers, responseError);
-                            return;
-                        }
+    [[session dataTaskWithRequest:request
+                completionHandler:^(NSData *data,
+                                    NSURLResponse *response,
+                                    NSError *error) {
+                    NSArray *turnServers = [NSArray array];
+                    if (error) {
+                        completionHandler(turnServers, error);
+                        return;
+                    }
+                    NSError *jsonError = nil;
+                    NSDictionary *dict = [NSJSONSerialization JSONObjectWithData:data
+                                                                         options:NSJSONReadingMutableContainers
+                                                                           error:&jsonError];
+                    if (jsonError) {
+                        NSError *responseError =
+                        [[NSError alloc] initWithDomain:[[RestCommClient sharedInstance] errorDomain]
+                                                   code:ERROR_WEBRTC_TURN
+                                               userInfo:@{
+                                                          NSLocalizedDescriptionKey: [NSString stringWithFormat:@"Error parsing JSON containing TURN servers. Check your TURN service"],
+                                                          }];
+                        completionHandler(turnServers, responseError);
+                        return;
+                    }
+                    
+                    NSObject * status = [dict objectForKey:@"s"];
+                    if (([status isKindOfClass:NSNumber.class] && [(NSNumber *)status integerValue] == 200)||
+                        ([status isKindOfClass:NSString.class] && [(NSString *)status isEqualToString:@"ok"])){
                         
                         NSArray * iceServers = [[dict objectForKey:@"d"] objectForKey:@"iceServers"];
-                        //turnServers = [RTCICEServer serversFromCEODJSONDictionary:dict];
+                        //for v3, ice servers are "v"
+                        if (!iceServers){
```

Comment:
I'd say instead of falling back to "v" if "d" key doesn't return any ice servers, pass as an argument to requestServersWith*() whether V2 or V3 is used and decide based on that which key to check (i.e. "v" or "d"), so that it's clearer and error reporting can be more helpful, because if the designated key is not there we knot that it's an error

Reply:
done.